src/session.rs

```diff
@@ -671,7 +720,7 @@ where
             .await
         {
             Ok(new_session_response) => Ok(Client {
-                tx,
+                tx: client.tx,
                 is_legacy: false,
                 new_session_response: Some(wd::NewSessionResponse::from_wd(new_session_response)),
             }),
```

Comment:
I think instead of creating a new `Client` here, you can just do `client.new_session_response = Some(..)` and then return `client`.

Reply:
Updated it to set `client.new_session_response` directly before returning.